web/src/pages/playground.tsx

```diff
@@ -0,0 +1,253 @@
+
+    const NumberInput = ({field, label, min, max, parser=(v) => parseInt(v, 10)} : InputField) => (
+        <>
+            <label for={field} className="col-span-3 inline-block">{label}: </label>
+            <input name={field} value={settings[field]} className="w-20"
+                   onChange={between(field, min && parser(min), max && parser(max), parser)}
+                   type="number"
+            />
+        </>
+    )
+    const Slider = ({field, label, min=0, max=1, step=0.01, parser=parseFloat}: InputField) => (
+        <>
+            <label for={field} className="col-span-2">{label}:</label>
+            <input name={field}  className="col-span-2"
+                   value={settings[field]}
+                   onChange={floatBetween(field, parser(min), parser(max))}
+                   type="range" min={min} max={max} step={step}
+            />
+        </>
+    )
+
+    return (
+        <div className="chat-settings w-[400px] flex-none outline-black border-2 mx-5 grid grid-cols-4 gap-4"
+             style={{height: "fit-content"}}>
+            <SectionHeader text="Models"/>
+            <label for="completions-model" className="col-span-2">Completions model:</label>
+            <select name="completions-model" className="col-span-2" value={settings.completions} onChange={update('completions')}>
+                {COMPLETION_MODELS.map((name) => (
+                    <option value={name}>{name}</option>
+                ))}
+            </select>
+
+            <label for="encoder" className="col-span-2">Encoder:</label>
+            <select name="encoder" className="col-span-2" value={settings.encoder} onChange={update('encoder')}>
+                {ENCODERS.map((name) => (
+                    <option value={name}>{name}</option>
+                ))}
+            </select>
+
+            <SectionHeader text="Token options"/>
+            <NumberInput field="numTokens" label="Tokens" min="1"/>
+            <NumberInput field="tokensBuffer" label="Number of tokens to leave as a buffer when calculating remaining tokens" min="0" max={settings.tokensBuffer} />
+
+            <SectionHeader text="Prompt options"/>
+            <NumberInput field="topKBlocks" label="Number of blocks to use as citations" min="1"/>
+            <NumberInput field="maxHistory" label="The max number of previous interactions to use" min="0"/>
+
+            <Slider field="contextFraction" label="Approximate fraction of num_tokens to use for citations text before truncating"/>
+            <Slider field="historyFraction" label="Approximate fraction of num_tokens to use for history text before truncating"/>
+        </div>
+    )
+}
+
+type ChatPromptParams = {
+    settings: LLMSettings,
+    query: string,
+    history: Entry[],
+    updateSettings: (updater: (settings: LLMSettings) => LLMSettings) => void,
+}
+
+const ChatPrompts = ({settings, query, history, updateSettings}: ChatPromptParams) => {
```

Comment:
Displays the prompt that will be used. This is pretty much what will be sent to the LLM, apart from the citations chunks which first have to be fetched. So this panel allows one to pretty much pass in arbitrary prompts:

![image](https://github.com/StampyAI/stampy-chat/assets/3942390/416476f5-e9e4-4302-ba26-b527825c30e9)